### PR TITLE
Add FTPHook

### PR DESCRIPTION
**Pull request by jlowin**
I wrote this based on @martingrayson's implementation in #62 since it was never merged. Adding it here as a PR in response to #41 and #399.

### CODE REVIEW

**Comment by martingrayson** [comment]
Sweet! Good job @jlowin!

**Comment by jlowin** [comment]
@martingrayson most of the credit is yours!

**Comment by mistercrunch** [comment]
Can we move the hook to `airflow/contrib/hooks/` since there are no unit tests and I can't really test it here in the absence of an FTP server?

**Comment by jlowin** [comment]
@mistercrunch no problem

File: airflow/hooks/__init__.py [hunk 1]
```diff
@@ -27,6 +27,7 @@
 }
 
 _import_module_attrs(globals(), _hooks)
+from airflow.contrib.hooks import *
```
Comment (mistercrunch) [inline airflow/hooks/__init__.py]: Let's leave this line out until we retrofit `import_module_attrs` to play nicely with `contrib/`, or actually, people can explicitly reference `airflow.contrib.*` in their script, makes it more clear that it is community contributed.

Comment (jlowin) [inline airflow/hooks/__init__.py]: @mistercrunch got it. It's now accessible from `airflow.contrib.hooks.FTPHook` and I added a placeholder module to show to how to add operators to contrib.

**Comment by mistercrunch** [comment]
The only thing missing thing is a sarcastic `logging.warning` message about using FTP in 2015 :)

**Comment by jlowin** [comment]
@mistercrunch What a wonderful world this would be...

**Pull request merged**